src/osx/cocoa/dataview.mm

```diff
@@ -2910,16 +2912,45 @@ void UpdateWithRow(int row)
         }
     }
 
+    if ( attr.HasBackgroundColour() && [cell backgroundStyle] == NSBackgroundStyleLight )
+    {
+        // Only NSTextFieldCell provides a method for setting the background colour,
+        // so check that this method is available before using it
```

Comment:
I'd also avoid repeating the same comment and just say "the logic here is the same as for text colour above".

Reply:
I usually prefer comments like this to standalone. Since these two pieces exist for different functionality, I would be afraid that if someone goes in and modifies the text color section in some way, this comment could make no sense anymore.

Reply:
I'm more afraid of the fact that someone will modify one of those comments but not the other one and they will become different for nor good reason and then (much) later some poor soul will have no idea which one is correct. And this does tend to happen in practice...

To ensure that this code/comment is modified if the text colour one is, you could also add "Note that we do the same thing for the background colour below". But I think that in practice this will be pretty obvious anyhow, _especially_ if they're both inside the background style test (as per the previous comment).